v2/observer/jobs.go

```diff
@@ -76,10 +87,13 @@ func (o *observer) syncJobPod(obj interface{}) {
 		status.Phase = core.JobPhaseRunning
 	case corev1.PodSucceeded:
 		status.Phase = core.JobPhaseSucceeded
+		cancelTimer()
 	case corev1.PodFailed:
 		status.Phase = core.JobPhaseFailed
+		cancelTimer()
 	case corev1.PodUnknown:
 		status.Phase = core.JobPhaseUnknown
+		cancelTimer()
 	}
```

Comment:
Besides invoking `cancelTimer()`, we also need to remove the `cancelTimer` func from the map or else we'll slowly leak memory by indefinitely retaining a pointer to every cancel function for every job.

Reply:
Probably the best thing to do is make the `cancelTimer` function remove _itself_ from the map when it exits.

Reply:
Oh yes, thank you for identifying.